packages/application/src/tokens.ts

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { CommandRegistry } from '@phosphor/commands';
+
+import { ServerConnection, ServiceManager } from '@jupyterlab/services';
+
+import { ReadonlyJSONObject, Token } from '@phosphor/coreutils';
+
+import { IDisposable } from '@phosphor/disposable';
+
+import { ISignal } from '@phosphor/signaling';
+
+/**
+ * A token for which a plugin can provide to respond to connection failures
+ * to the application server.
+ */
+export const IConnectionLost = new Token<IConnectionLost>(
+  '@jupyterlab/apputils:IConnectionLost'
+);
+
+export type IConnectionLost = (
```

Comment:
This token interface is a function. I am not aware of anywhere else where we do this, but it seemed reasonable to me.

Reply:
I'm fine with this file as-is